### ISSUE TITLE
Spark GPU support

### ISSUE THREAD

**Issue by akhurana001** [lyft/flyte#224]
**From a slack request:** 

**Motivation: Why do you think this is important?**
Spark 2.4.x doesn't natively support GPU but  SparkOperator enables this via a mutating webhook. Explore if Flyte can add support for GPU for spark tasks using this webhook.

**Goal: What should the final outcome look like, ideally?**
Users can specify gpu resources for Spark driver/executors and execute spark jobs on GPU nodes.

**Describe alternatives you've considered**

**Flyte component**
- [x] FlyteIDL (Flyte specification language)
- [x] Flytekit (Python SDK)
- [x] FlytePlugins

**[Optional] Propose: Link/Inline**
This would need some changes in flyteidl/flytekit/flyteplugins to pass gpu config through to the spark-operator. 

**Additional context**

**Comment by kumare3** [comment]
@akhurana001 I think this is done right? Should we close it?

**Comment by akhurana001** [comment]
> @akhurana001 I think this is done right? Should we close it?

Not eally, we would need to plug it through in Idl/flytekit/plugins to populate it in the SparkApplication CRD post which the webhook  will enforce it.

**Comment by akhurana001** [comment]
Spark 3.0 went GA a couple of weeks ago and natively supports requesting GPU resources : 
```
spark.driver.resource.{resourceName}.vendor
spark.executor.resource.{resourceName}.vendor
```
I think we can avoid adding explicit gpu support via webhook given its officially supported by Spark 3.0 now.

**Comment by kumare3** [comment]
Agreed, and Spark3.0 support has been merged v0.6.0
